lib/awful/mouse/snap.lua

```diff
@@ -20,6 +21,54 @@ local capi = {
 
 local module = {}
 
+local function detect_screen_edges(c, snap)
+    local coords = capi.mouse.coords()
+
+    local sg = c.screen.geometry
+
+    local v, h = nil
+
+    if math.abs(coords.x) <= snap + sg.x and coords.x >= sg.x then
+        h = "left"
+    end
+
+    if math.abs((sg.x + sg.width) - coords.x) <= snap then
+        h = "right"
+    end
+
+    if math.abs(coords.y) <= snap + sg.y and coords.y >= sg.y then
+        v = "top"
+    end
+
+    if math.abs((sg.y + sg.height) - coords.y) <= snap then
+        v = "bottom"
+    end
+
+    return v, h
+end
+
+local current_snap = nil
```

Comment:
Hrm. I don't really know how, but can we somehow get rid of this global variable? This is used as a magic way to get some argument from the move callback below into the leave callback...